src/dso_api/dynamic_api/views/api.py

```diff
@@ -122,13 +64,62 @@ class DynamicApiViewSet(
     # Make sure composed keys like (112740.024|487843.078) are allowed.
     # The DefaultRouter won't allow '.' in URLs because it's used as format-type.
     lookup_value_regex = r"[^/]+"
+    lookup_url_kwarg = "pk"
 
     # Custom permission that checks amsterdam schema auth settings
     permission_classes = [permissions.HasOAuth2Scopes]
 
     # The 'bronhouder' of the associated dataset
     authorization_grantor: str = None
 
+    def initial(self, request, *args, **kwargs):
+        super().initial(request, *args, **kwargs)
+        self.temporal = TemporalTableQuery(request, self.model.table_schema())
+
+    @cached_property
+    def lookup_field(self):
+        """Overwritten from GenericAPIView to filter on temporal identifier instead."""
+        if self.temporal.is_versioned:
+            return self.model.table_schema().identifier[0]
```

Comment:
Same here. Why only the first element of `identifier`?

See also https://github.com/Amsterdam/dso-api/pull/389/files#r742560983

Reply:
It's the same behavior from the old code. I've added a comment about it:

`# Take the first field as grouper from ["identificatie", "volgnummer"]`